src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs

```diff
@@ -1548,6 +1549,45 @@ public C(string s)
 }}", index: 2);
         }
 
+        [WorkItem(51338, "https://github.com/dotnet/roslyn/issues/51338")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestSpecialStringCheck3()
+        {
+            var culture = CultureInfo.CurrentUICulture;
+            try
+            {
+                CultureInfo.CurrentUICulture = new CultureInfo("de-DE", useUserOverride: false);
+
+                await TestInRegularAndScript1Async(
+    @"
+using System;
+
+class C
+{
+    public C([||]string s)
+    {
+    }
+}",
+    @$"
+using System;
+
+class C
+{{
+    public C(string s)
+    {{
+        if (string.IsNullOrEmpty(s))
+        {{
+            throw new ArgumentException($""{string.Format(FeaturesResources._0_cannot_be_null_or_empty, "{nameof(s)}").Replace("\"", "\\\"")}"", nameof(s));
```

Comment:
Some languages (e.g zh-Hans), uses `”` instead of `"`. Do we need to specify a string comparison to prevent `”` from being incorrectly escaped? I'm not sure as I'm not of much awareness of globalization things, but noting this anyway in case it's a valid.

Reply:
the issue is us actualy inserting a literal `"` into the c# code, as that will terminate an in-progress string literal.  If otehr langauges uses *other* quote characters, it's not an issue as they won't terminate the string literals.